From 7a61fab4ef7e8971f5814db473ed34534e7b9585 Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Thu, 6 Jul 2017 14:42:27 -0500
Subject: [PATCH 228/278] drivers/misc: Update Axxia PEI Configuration

- In configuration 2, set sw_port_1 to 0x2.
- Always set the number of PHYs to 4 for 5600
  and 1 for 6700.

Signed-off-by: John Jacques <john.jacques@intel.com>
---
 drivers/misc/axxia-pei.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/drivers/misc/axxia-pei.c b/drivers/misc/axxia-pei.c
index 6d417aa..1ce29b2 100644
--- a/drivers/misc/axxia-pei.c
+++ b/drivers/misc/axxia-pei.c
@@ -1367,9 +1367,10 @@ static void enable_srio_lanes(enum SrioMode mode)
 		*/
 
 		set_sw_port_config0(pc0_PEI0x2_PEI2x2);
-		set_sw_port_config1(pc1_PEI1x2_PEI2x2);
+		set_sw_port_config1(pc1_PEI1x2_SATA0x1_SATA1x1);
 		set_pei0_rc_mode(rc_mode);
 		set_pipe_port_sel(pp_0_1_2_3);
+		set_pipe_nphy(four_phy);
 
 		if (pei0_mode)
 			release_reset(0);
@@ -1397,7 +1398,7 @@ static void enable_srio_lanes(enum SrioMode mode)
 		set_sw_port_config0(pc0_PEI0x2_SRIO0x2);
 		set_sw_port_config1(pc1_PEI1x2_PEI2x2);
 		set_pipe_port_sel(pp_0_1_2_3);
-		set_pipe_nphy(three_phy);
+		set_pipe_nphy(four_phy);
 
 		if (pei0_mode)
 			release_reset(0);
@@ -1451,6 +1452,7 @@ static void enable_srio_lanes(enum SrioMode mode)
 		set_sw_port_config0(pc0_SRIO1x2_SRIO0x2);
 		set_sw_port_config1(pc1_PEI1x2_PEI2x2);
 		set_pipe_port_sel(pp_0_1_2_3);
+		set_pipe_nphy(four_phy);
 
 		if (pei1_mode)
 			release_reset(2);
@@ -1521,6 +1523,7 @@ static void enable_srio_lanes(enum SrioMode mode)
 		set_sw_port_config0(pc0_SRIO1x2_SRIO0x2);
 		set_sw_port_config1(pc1_PEI1x4);
 		set_pipe_port_sel(pp_0_1_2_3);
+		set_pipe_nphy(four_phy);
 
 		if (pei1_mode) {
 			release_reset(2);
@@ -1614,11 +1617,9 @@ static void enable_srio_lanes(enum SrioMode mode)
 		/*
 		 * PEI0x2
 		 */
-		if (0) {
-			set_pei0_rc_mode(rc_mode);
-			set_pipe_port_sel(pp_0);
-			set_pipe_nphy(one_phy);
-		}
+		set_pei0_rc_mode(rc_mode);
+		set_pipe_port_sel(pp_0);
+		set_pipe_nphy(one_phy);
 
 		if (pei0_mode)
 			release_reset(0);
-- 
1.8.5.2.233.g932f7e4

